datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/mutate/BatchAddTagsResolver.java

```diff
@@ -127,9 +126,15 @@ private Boolean attemptBatchAddTagsWithSiblings(
     }
   }
 
-  private void validateTags(@Nonnull OperationContext opContext, List<Urn> tagUrns) {
+  private void validateTags(@Nonnull QueryContext context, List<Urn> tagUrns) {
     for (Urn tagUrn : tagUrns) {
-      LabelUtils.validateLabel(opContext, tagUrn, Constants.TAG_ENTITY_NAME, _entityService);
+      LabelUtils.validateLabel(
+          context.getOperationContext(), tagUrn, Constants.TAG_ENTITY_NAME, _entityService);
+
+      if (!LabelUtils.isAuthorizedToAssociateTag(context, tagUrn)) {
```

Comment:
Auth calls should come first before sending validation operations to the entity service, ideally done as soon as possible before any business logic occurs to short circuit.